edx_toggles/toggles/internal/waffle/legacy.py

```diff
@@ -0,0 +1,175 @@
+"""
+from abc import ABC
+
+from edx_django_utils.monitoring import set_custom_attribute
+
+from .flag import WaffleFlag as NewWaffleFlag
+from .switch import WaffleSwitch as NewWaffleSwitch
+
+
+class BaseNamespace(ABC):
+    """
+    A base class for a request cached namespace for waffle flags/switches.
+
+    An instance of this class represents a single namespace
+    (e.g. "course_experience"), and can be used to work with a set of
+    flags or switches that will all share this namespace.
+    """
+
+    def __init__(self, name, log_prefix=None):
+        """
+        Initializes the waffle namespace instance.
+
+        Arguments:
+            name (String): Namespace string appended to start of all waffle
+                flags and switches (e.g. "grades")
+            log_prefix (String): Optional string to be appended to log messages
+                (e.g. "Grades: "). Defaults to ''.
+
+        """
+        set_custom_attribute(
+            self.__class__.__module__,
+            "{}[{}]".format(self.__class__.__name__, name),
+        )
+        assert name, "The name is required."
+        self.name = name
+        self.log_prefix = log_prefix if log_prefix else ""
+
+    def _namespaced_name(self, toggle_name):
+        """
+        Returns the namespaced name of the waffle switch/flag.
+
+        For example, the namespaced name of a waffle switch/flag would be:
+            my_namespace.my_toggle_name
+
+        Arguments:
+            toggle_name (String): The name of the flag or switch.
+        """
+        return "{}.{}".format(self.name, toggle_name)
+
+
+class WaffleSwitchNamespace(BaseNamespace):
+    """
+    Legacy waffle switch namespace class.
+    """
+
+    def is_enabled(self, switch_name):
+        """
+        Returns whether or not the switch is enabled.
+        """
+        return NewWaffleSwitch(self._namespaced_name(switch_name)).is_enabled()
+
+    def set_request_cache_with_short_name(self, switch_name, value):
+        """
+        Explicitly set request cache value for the (non-namespaced) switch name. You should avoid using this method as
+        much as possible as it may have unexpected side-effects.
+        """
+        namespaced_name = self._namespaced_name(switch_name)
+        # pylint: disable=protected-access
+        NewWaffleSwitch(namespaced_name)._cached_switches[namespaced_name] = value
+
+
+class WaffleSwitch(NewWaffleSwitch):
+    """
+    Represents a single waffle switch, enhanced with request level caching.
+    """
+
+    def __init__(self, waffle_namespace, switch_name, module_name=None):
+        if not isinstance(waffle_namespace, str):
+            waffle_namespace = waffle_namespace.name
+        set_custom_attribute("deprecated_edx_toggles_waffle", "WaffleSwitch")
+
+        # Non-namespaced flag_name attribute preserved for backward compatibility
+        self.switch_name = switch_name
```

Comment:
Related to an earlier discussion about protecting against backward-incompatible changes, can we add a property for this, and then add a custom attribute for this and for all the deprecated methods that will be going away.  

We can either go with something like:
```
deprecated_waffle_legacy_method=WaffleSwitch[some_name].switch_name
```
Or
```
deprecated_waffle_legacy_switch_name=some_name
```
It would be easier to query for the former, static key name, but it means we just get the last value, so it could be more like an onion, where we solve one issue and then find the next.

Reply:
Sure, I can do either. Do you prefer we go with the former?

Reply:
@regisb: We can absolutely follow-up with this if you want to land this first. ^  Just need it before 2.0.0.  I still need to double-check this PR once I am done with meetings.

Reply:
Please let me know if you agree with the changes I pushed.